thunder/torch/__init__.py

```diff
@@ -1503,6 +1503,39 @@ def exp2_(a):
     return prims.copy_(exp2(a), a)
 
 
+# fake out of place variant
+@torchsymbol(id="exponential")
+def exponential(a, rate=1, generator=None):
+    assert generator is None
+    utils.check(
+        not thunder.dtypes.is_complex_dtype(a.dtype)
```

Comment:
This could affirmatively check that `a` is a floating point type instead of asserting it's not any other type, which would probably be more straightforward